main/test/ca/uwaterloo/flix/language/feature/Test.Expression.Match.Array.flix

```diff
@@ -0,0 +1,111 @@
+/*
+@test
+def arrayPatternEmpty(): Bool = match [] {
+case [] => true
+case _ => false
+}
+*/
+
+@test
+def arrayPatternUnit1(): Bool = match [()] {
+case [()] => true
+case _ => false
+}
+
+@test
+def arrayPatternInt1(): Bool = match [1] {
```

Comment:
I would propose a different naming scheme more in line with what we currently have:

- matchUnitArray01 etc.
- matchInt32Array01 etc.
- matchFloat64Array01 etc.
- matchInt32ArrayArray01

One thing that should definitely be fixed is that Flix does not use the names "byte" or "short". The names are: Int8, Int16, Int32, Int64, Float32, Float64.